games/rummy/scenes/PlayScene.ts

```diff
@@ -0,0 +1,1160 @@
+import { Result } from '@/games/common/types/game';
+import Card from '../../common/Factories/cardImage';
+import Deck from '../../common/Factories/deckImage';
+import GAME from '../../common/constants/game';
+import STYLE from '../../common/constants/style';
+import GamePhase from '../constants/gamePhase';
+import GameResult from '../constants/gameResult';
+import RummyPlayer from '../models/rummyPlayer';
+
+import Table from '../../common/Factories/tableScene';
+import Text = Phaser.GameObjects.Text;
+import Zone = Phaser.GameObjects.Zone;
+import TimeEvent = Phaser.Time.TimerEvent;
+
+export default class PlayScene extends Table {
+  // TODO: private->#に変更する
+  private playerDecks: Array<Deck> = []; // player, house
+
+  private deckSizeTexts: Array<Text> = []; // player, house
+
+  private dropZones: Array<Zone> = []; // player, house
+
+  private dropCardRanks: Array<number> = []; // player, house
+
+  private timeEvent: TimeEvent | undefined;
+
+  private housePlayTimeEvent: TimeEvent | undefined;
+
+  private houseDealTimeEvent: TimeEvent | undefined;
+
+  #countDownSound: Phaser.Sound.BaseSound | undefined;
+
+  private meldZone: Zone | undefined;
+
+  private meldZoneEach: Zone[] = [];
+
+  private meldZoneEachCounter = 0; // 新しいcardが追加されたら、増える。
+
+  private frontDeckZone: Zone | undefined;
+
+  private backDeckZone: Zone | undefined;
+
+  private isPlayerTurn = false; // 初期状態ではハウスのターンとします
+
+  private houseFlag = false;
+
+  private playerFlag = false;
+
+  private deckBack: Deck[] = [];
```

Comment:
deckは複数存在しないので、配列にする必要はないのではないでしょうか？


```suggestion
  private deckBack: Deck;
```